.NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/TimexResolver.cs

```diff
@@ -236,18 +236,35 @@ private static List<Resolution.Entry> ResolveDuration(TimexProperty timex)
 
         private static Tuple<string, string> YearWeekDateRange(int year, int weekOfYear, bool? isWeekend)
         {
-            var dateInWeek = new DateObject(year, 1, 1) + TimeSpan.FromDays((weekOfYear - 1) * 7);
+            var firstMondayInWeek = FirstDateOfWeek(year, weekOfYear, System.Globalization.CultureInfo.InvariantCulture);
 
             var start = (isWeekend == null || isWeekend.Value == false) ?
-                            TimexDateHelpers.DateOfLastDay(DayOfWeek.Monday, dateInWeek) :
-                            TimexDateHelpers.DateOfNextDay(DayOfWeek.Saturday, dateInWeek);
-            var end = TimexDateHelpers.DateOfLastDay(DayOfWeek.Monday, dateInWeek + TimeSpan.FromDays(7));
+                            firstMondayInWeek :
+                            TimexDateHelpers.DateOfNextDay(DayOfWeek.Saturday, firstMondayInWeek);
+            var end = firstMondayInWeek + TimeSpan.FromDays(7);
 
             return new Tuple<string, string>(
                 TimexValue.DateValue(new TimexProperty { Year = start.Year, Month = start.Month, DayOfMonth = start.Day }),
                 TimexValue.DateValue(new TimexProperty { Year = end.Year, Month = end.Month, DayOfMonth = end.Day }));
         }
 
+        // this is based on https://stackoverflow.com/questions/19901666/get-date-of-first-and-last-day-of-week-knowing-week-number/34727270
+        private static DateObject FirstDateOfWeek(int year, int weekOfYear, System.Globalization.CultureInfo cultureInfo)
+        {
+            // ISO uses FirstFourDayWeek, and Monday as first day of week, according to https://en.wikipedia.org/wiki/ISO_8601
+            var jan1 = new DateObject(year, 1, 1);
+            int daysOffset = (int)DayOfWeek.Monday - (int)jan1.DayOfWeek;
+            var firstWeekDay = jan1.AddDays(daysOffset);
+
+            int firstWeek = cultureInfo.Calendar.GetWeekOfYear(jan1, System.Globalization.CalendarWeekRule.FirstFourDayWeek, DayOfWeek.Monday);
+            if ((firstWeek <= 1 || firstWeek >= 52) && daysOffset >= -3)
```

Comment:
Can you also add a unit test to show that W53 works as intended in years that do have it?
Perhaps two test cases, one with reference date at the very end of a year and one with a reference date at the very beginning, but that will still map back to the previous year week 53.

Reply:
Those missing tests somehow got removed when I updated my local repo - sorry about that - re-added them now. I also put in the new tests you suggested - that what you had in mind?

Reply:
Seems good. We'll merge once the build passes. Thanks!

Reply:
great stuff, thanks!